webapp/actions/post_actions.jsx

```diff
@@ -252,3 +252,37 @@ export function loadProfilesForPosts(posts) {
 
     AsyncClient.getProfilesByIds(list);
 }
+
+export function createPost(post, doLoadPost) {
+    Client.createPost(post,
+        (data) => {
+            if (doLoadPost) {
+                loadPosts(post.channel_id);
+            } else {
+                PostStore.removePendingPost(post.pending_post_id);
+            }
+
+            AppDispatcher.handleServerAction({
+                type: ActionTypes.RECEIVED_POST,
+                post: data
+            });
+        },
+
+        (err) => {
+            if (err.id === 'api.post.create_post.root_id.app_error') {
+                // this should never actually happen since you can't reply from this textbox
+                this.showPostDeletedModal();
```

Comment:
I might need some help here to understand where I need to add the showPostDeletedModal(); @enahum